### ISSUE TITLE
Laura Ramirez: Incremental Development

### ISSUE THREAD

**Issue by lauraramirez220612** [HackYourFutureBelgium/class-15#96]
## Suggested Study
**Incremental Development**
- [x] [What is this?](https://www.youtube.com/watch?v=GzzkpAOxHXs)
- [x] [Agile Development by Cartoon](https://www.youtube.com/watch?v=Z9QbYZh1YXY&list=PLBUu5aGDLKnbeEx8U-5r436bw6p9wv1rS)
- [x] [Atlassian](https://www.atlassian.com/team-playbook/plays), [Team Playbook](https://www.atlassian.com/team-playbook/plays)
- [x] [Splitting User Stories](https://www.youtube.com/watch?v=EDT0HMtDwYI)
- [x] [Three key parts](https://www.youtube.com/watch?v=ctFzjMygaRo)

**Working Together**
- [x] Managing Conflict](https://buildtogether.tech/conflict/)

**Git & GitHub**
- [x] [learngitbranching](https://learngitbranching.js.org)
- [x] [git & github for poets](https://www.youtube.com/watch?v=BCQHnlnPusY&list=PLRqwX-V7Uu6ZF9C0YMKuns9sLDzK6zoiV)
- [x] lab.github.com:
     - [x] [First Day on GitHub](https://lab.github.com/githubtraining/first-day-on-github)
     - [x] [First Week on GitHub](https://lab.github.com/githubtraining/first-week-on-github)
- [ ] [The Net Ninja](https://www.youtube.com/watch?v=QV0kVNvkMxc&list=PL4cUxeGkcC9goXbgTDQ0n_4TBzOO0ocPR&index=9)
- [ ] [git-it](https://github.com/jlord/git-it-electron/)
- [ ] [Understand how to use Atomic Commits](https://curiousprogrammer.io/blog/how-to-craft-your-changes-into-small-atomic-commits-using-git/)

**Collaborating on GitHub**
- [ ] [Adding collaborators to a repository](https://www.youtube.com/watch?v=p49LRx3hYI8)
- [ ] [about code reviews](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews)
- [ ] [requesting a code review](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review)
- [ ] [Git Workflow for 2](https://github.com/hackyourfuturebelgium/git-workflow-workshop-for-two)
- [ ] [Pull Requests](https://www.youtube.com/watch?v=2M16faxEQsg)
- [ ] [Git & GitHub for Poets](https://www.youtube.com/watch?v=BCQHnlnPusY&list=PLRqwX-V7Uu6ZF9C0YMKuns9sLDzK6zoiV)
- [ ] The Net Ninja: [11](https://www.youtube.com/watch?v=MnUd31TvBoU&list=PL4cUxeGkcC9goXbgTDQ0n_4TBzOO0ocPR&index=12)
- [ ] linking PRs to Issues: [reference 1](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue), [reference 2](https://docs.github.com/en/github/writing-on-github/autolinked-references-and-urls)
- [ ] [closing Issues using keywords](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords)

**Command Line Interface**
- [ ] [Jesse Showalter](https://www.youtube.com/watch?v=5XgBd6rjuDQ)
    - [ ] CLI games:
       - [ ] [bashcrawl](https://gitlab.com/slackermedia/bashcrawl/) - clone & play
       - [ ] [Terminus](https://web.mit.edu/mprat/Public/web/Terminus/Web/main.html) - online
       - [ ] [iTerm](https://shubhamrath.dev/lterm/) - online
- [ ] [A huge chat sheet](https://gist.github.com/LeCoupa/122b12050f5fb267e75f)
- [ ] [study.hyf.be](https://study.hackyourfuture.be/cli)

**Writing READMEs**
- [ ] [makeareadme.com](https://www.makeareadme.com)
- [ ] [bulldogjob](https://bulldogjob.com/news/449-how-to-write-a-good-readme-for-your-github-project)
- [ ] [meakaakka](https://meakaakka.medium.com/a-beginners-guide-to-writing-a-kickass-readme-7ac01da88ab3)

**DevTools and the DOM**
- [ ] [How to inspect an element](https://www.lifewire.com/get-inspect-element-tool-for-browser-756549)
- [ ] [inspecting-the-dom](https://hackyourfuturebelgium.github.io/inspecting-the-dom/)
- [ ] [Modify the DOM. (does the source change?)](https://zapier.com/blog/inspect-element-tutorial/)

**HTML & CSS**
- [ ] FCC, Responsive Web Design: [exercises](https://www.freecodecamp.org/learn/), [video](https://www.youtube.com/watch?v=srvUrASNj0s)
- [ ] [mmtuts: HTML & CSS](https://www.youtube.com/watch?v=TKYsuU86-DQ&list=PL0eyrZgxdwhwNC5ppZo_dYGVjerQY3xYU)
- [ ] [CSS Games](https://study.hackyourfuture.be/html-css/css#games-to-learn-css)
- [ ] 🥚 [html-css-git-exercises](https://github.com/hackyourfuturebelgium/html-css-git-exercises)
- [ ]  🐣 [HTML-CSS-Practice-Problems](https://github.com/DevMountain/HTML-CSS-Practice-Problems)
- [ ]  🐥 [css-exercises](https://github.com/drwpow/css-exercises)

**User Stories**
- [ ] [User Stories](https://www.youtube.com/watch?v=apOvF9NVguA)
- [ ]  Acceptance Criteria
      - [ ] [Yodiz](https://www.yodiz.com/blog/user-stories-acceptance-definition-and-criteria-in-agile-methodologies/)
      - [ ] [Zepel](https://zepel.io/agile/acceptance-criteria-for-user-stories/)
      - [ ] [Ruby Garage](https://rubygarage.org/blog/clear-acceptance-criteria-and-why-its-important)
      - [ ] [The Infinity Project](https://www.youtube.com/watch?v=KYS0ptJ4JWc)
- [ ] Setting Priorities
      - [ ] [Many Strategies](https://zapier.com/blog/how-to-prioritize/)
      - [ ] The Priority Quadrant
  [The Quadrant](https://www.youtube.com/watch?v=NGvsxPOmWuw)
  [Problems with this method](https://www.linkedin.com/pulse/why-prioritization-impacteffort-doesnt-work-itamar-gilad)
      - [ ] Other Perspectives
          - [ ]    [Alex Ponomarev](https://medium.com/swlh/prioritizing-user-stories-in-agile-projects-d1dd8dd79165)
          - [ ]    [Michael Lant](https://michaellant.com/2010/05/21/how-to-easily-prioritize-your-agile-stories/)

## Exercises

- [ ] [repo](https://....    ) (with a complete README)
- [ ] [live demo](https://....)
- [ ] [development strategy](https://...)
- [ ] [one branch per step](https://...)
- [ ] [one closed PR per step](https://...)
---

### Blocked
Sometimes with GitHub and Terminal,  specially Branch, Push, Pull, and to understand the workflow.

### Next Steps
Work on the code to recreate Lokuri website.

### Tip(s) of the week
* Learn how to plan the project,  this assignment with the help of my classmate Senait.
* Make notes.
* Focus and understand better from GitHub Tutorials and then with help of my classmate Axil.

### Prepared for Sunday
- [ ] Sunday Prep

---

-->

**Comment by colevandersWands** [comment]
@lauraramirez220612 , @hserdogan94 , it looks like you both wrote the same check-in? #97

**Comment by lauraramirez220612** [comment]
Hello Evan, 
Please tell me if it is that you have required to me,  I wrote this on Wednesday morning:

## Week 1
## Exercises
- [ ] [repo](https://....    ) (with a complete README)
- [ ] [live demo](https://....)
- [ ] [development strategy](https://...)
- [ ] [one branch per step](https://...)
- [ ] [one closed PR per step](https://...)
---

### Blocked
Sometimes with GitHub and Terminal,  specially Branch, Push, Pull, and to understand the workflow.

### Next Steps
Work on the code to recreate Lokuri website.

### Tip(s) of the week
* Learn how to plan the project,  this assignment with the help of my classmate Senait.
* Make notes.
* Focus and understand better from GitHub Tutorials and then with help of my classmate Axil.

### Prepared for Sunday
- [ ] Sunday Prep

**Comment by colevandersWands** [comment]
👋 @lauraramirez220612 , this is perfect!  I was a little extra confused because I saw @hserdogan94  had the exact same check-in and I wasn't sure who had copied who.  Now I see it was him who copied you.

Happy to hear you've found some good classmates  : )

> Sometimes with GitHub and Terminal, specially Branch, Push, Pull, and to understand the workflow.

Is there any commands or workflows in particular you're blocked on?  Let me know and we can be prepared to review them on Sunday

**Comment by lauraramirez220612** [comment]
Yes,   still 3 in my group are fighting with:
git pull upstream main or origin,    git push upstream...,   This morning we are trying but some conflicts appears.   We are going to start again as from first step.  But I could not finish my assignment for today to contribute with my group. 😔

**Comment by lauraramirez220612** [comment]
## Week 2 

#### I Needed Help With:
Git conflicts,  some classmates have helped me with,  I took notes in my notebook but I need to understand better why the conflicts and how to fix it out.

#### What went well?
Working with VSC and Terminal,  now I use more properly them (it makes me happy), although I need to investigate more in their menu and commands.   
I have learned from my classmates to use Excalidraw app.
I feel good with my group, they are very helpful.

#### What went less well?
Updating my repository in my local pc with the master origin main.
I still don't feel sure to make Pull Requests.

#### Lessons Learned


### Sunday Prep Work

**Comment by colevandersWands** [comment]
> The Websites architecture, all behind the scene, really interesting.

:)

**Comment by colevandersWands** [comment]
@lauraramirez220612, Apologies for missing your comment last week.  Next time if you mention me it'll show up in my notifications and I'll get back to you.

Have you tried the [git games](https://github.com/HackYourFutureBelgium/class-15/tree/master/chill-zone#git-games)? They can be helpful for understanding, it helps to visualize what's happening

**Comment by lauraramirez220612** [comment]
Yes,  the game is friendly,  tomorrow Saturday I will have more time to
focus on it.    I would like to share that today after my meeting with
Moamin who is helping me with Github,  *I really did! *💻 some changes in
my assignment ( git branch, git checkout, git add, git commit -m "  , git
push --set-upstream origin  and Open PR in remote and finally Create PR).
Thanks HYF.

**Comment by lauraramirez220612** [comment]
Week 3

I Needed Help With:

Recreation of HYF website,  some classmates have helped me with.

What went well?

Watching how my classmates work,  after Senait and Tatsiana meeting I got my ideas so clear and specially how to start the assignment.


What went less well?

When I finished the html and css files, when I checked the Preview some icons were in different place, I mean not the same layout,  Senait, Tatsiana and me were trying to fix it up and the failure was that I did wrong some closing </div>
This week for me was difficult to organise my time,  I had a lot of noise here at home because workers were making reparations in the bathroom.

Lessons Learned

html
css
Inspect a website


Sunday Prep Work

 Sunday Prep

**Comment by colevandersWands** [comment]
@lauraramirez220612 , happy to hear Senait and Tatsiana were so helpful : )  Are you feeling better about HTML & CSS?  You'll have two weeks to build an individual project starting today so that should give you some time to be comfortable.

**Comment by lauraramirez220612** [comment]
That's right Evan, I'll keep practicing for the individual project.  Thanks
